config/tests/samples/create/harness.go

```diff
@@ -491,44 +490,11 @@ func (t *Harness) waitForCRDReady(obj client.Object) {
 }
 
 func (h *Harness) CompareGoldenFile(p string, got string, normalizers ...func(s string) string) {
-	if os.Getenv("WRITE_GOLDEN_OUTPUT") != "" {
-		// Short-circuit when the output is correct
-		b, err := os.ReadFile(p)
-		if err == nil {
-			want := string(b)
-			for _, normalizer := range normalizers {
-				got = normalizer(got)
-				want = normalizer(want)
-			}
-			if want == got {
-				return
-			}
-		}
-
-		if err := os.WriteFile(p, []byte(got), 0644); err != nil {
-			h.Fatalf("failed to write golden output %s: %v", p, err)
-		}
-		h.Errorf("wrote output to %s", p)
-	} else {
-		want := string(h.MustReadFile(p))
-
-		for _, normalizer := range normalizers {
-			got = normalizer(got)
-			want = normalizer(want)
-		}
-
-		if diff := cmp.Diff(want, got); diff != "" {
-			h.Errorf("unexpected diff in %s: %s", p, diff)
-		}
-	}
+	test.CompareGoldenFile(h.T, p, got, normalizers...)
```

Comment:
Hmmm can we remove CompareGoldenFile and MustReadFile from harness? I do not see any reason we need them here, we can call the function via test utils.go

Reply:
I think it's better to keep it from the developer ergonomics point of view - easier discovery and one less argument to pass.